apps/dashboard/src/app/(main)/(protected)/projects/[projectId]/sign-up-rules/page-client.tsx

```
@@ -278,28 +425,18 @@ function RuleTriggerHistoryDialog({
+      if (nextRequestId !== latestRequestIdRef.current) return;
       setLoadingError(error instanceof Error ? error.message : "Failed to load triggers");
     } finally {
-      if (nextRequestId !== latestRequestIdRef.current) {
-        return;
-      }
-      if (reset) {
-        setIsInitialLoading(false);
-      } else {
-        setIsLoadingMore(false);
-      }
+      if (nextRequestId !== latestRequestIdRef.current) return;
```

Biome (2.4.11)

[error] Unsafe usage of 'return'.

(lint/correctness/noUnsafeFinally)